crossmatch/ztf_ps1_crossmatch.md

```diff
@@ -233,22 +233,22 @@ nworkers = [1, 2, 4, 8, 16, 32, 64, 128, 256, "default", "scale"]
 def plot_by_nworkers(num_cpus, ax):
     # Plot execution time vs. number of dask workers for each scale job
     b = benchmarks.loc[num_cpus]
-    
+
     for n in b.index.levels[0]:
         try:
             # get the relevant benchmarks
             b_N = b.xs(n, level="Nrows").reset_index()
 
             # convert Nworkers to categorical with desired order
-            b_N["Nworkers"] = pd.Categorical(b_N["Nworkers"], 
-                                             categories=[str(x) for x in nworkers], 
+            b_N["Nworkers"] = pd.Categorical(b_N["Nworkers"],
+                                             categories=[str(x) for x in nworkers],
                                              ordered=True)
             b_N = b_N.sort_values("Nworkers")
 
             # only plot if there is more than 1 data point
             if len(b_N) > 1:
                 ax.plot("Nworkers", "time", marker="s", linestyle="-", data=b_N, label=f"{n} rows")
-        except:
+        except KeyError:
             pass
```

Comment:
when does this happen, and when that happens, is that expected and normal?

Reply:
He is looping over dataframe indexes but not all of them have the data he's getting. I think it's "expected" and "normal" in that sense. But actually, I can move all but one of the code lines outside of the try/except. Pushing momentarily. It could probably be refactored to get rid of the try/except altogether but I'm not seeing exactly how to do that right now and not able to spend much more time on it today.